elements/howto-toggle-button/howto-toggle-button.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    SPACE: 32,
+    ENTER: 13,
+  };
+
+  /**
+   * Cloning contents from a &lt;template&gt; element is more performant
+   * than using innerHTML because it avoids addtional HTML parse costs.
+   */
+  const template = document.createElement('template');
+  template.innerHTML = `
+    <style>
+      :host {
+        background-color: #eee;
+        display: inline-block;
+        padding: 3px;
+        border: 1px solid #333;
+        border-radius: 3px;
+        cursor: default;
+        user-select: none;
+        transition: background-color .2s ease;
+      }
+      :host([hidden]) {
```

Comment:
I think `[hidden]` works out-of-the-box.

Reply:
I was trying to follow the best practices here: https://developers.google.com/web/fundamentals/architecture/building-components/best-practices  "In addition to a default display style, add support for hidden with :host([hidden]) { display: none }". Also, howto-checkbox seems to do the same. But, I can remove it if that's the recommended practice.

Reply:
Yeah unfortunately `:host { display: inline-block }` will trump the UA stylesheet's `[hidden]` style. So Custom Elements need to explicitly add support for hidden back in. I think what you're doing here is correct.